tests/Common/ComponentsTest.php

```diff
@@ -2699,4 +2696,34 @@ public function testCopyPreservesState()
         $configurationResponse = $components->getConfiguration('wr-db', $newConfig['id']);
         $this->assertEquals($state, $configurationResponse['state']);
     }
+
+    public function testRevertingConfigRowVersionWillNotCreateEmptyConfiguration()
+    {
+        $components = new \Keboola\StorageApi\Components($this->_client);
+        $configuration = new \Keboola\StorageApi\Options\Components\Configuration();
+        $componentId = 'wr-db';
+        $configurationId = 'main-1';
+        $configuration
+            ->setComponentId($componentId)
+            ->setConfigurationId($configurationId)
+            ->setName('Main');
+        $configurationRow = new ConfigurationRow($configuration);
+        $configurationRow->setChangeDescription('Test description');
+
+        $components->addConfiguration($configuration);
+        $rowArray = $components->addConfigurationRow($configurationRow);
+        $originalConfigurationArray = $components->getConfiguration($componentId, $configurationId);
+        $components->deleteConfigurationRow($componentId, $configurationId, $rowArray['id']);
+        $components->rollbackConfiguration($componentId, $configurationId, $originalConfigurationArray['version']);
+        $rollbackedConfigurationArray = $components->getConfiguration($componentId, $configurationId);
+
+        // we're not comparing the same row
+        $this->assertNotEquals($originalConfigurationArray['rows'][0]['version'], $rollbackedConfigurationArray['rows'][0]['version']);
+        $this->assertNotEmpty($rollbackedConfigurationArray['rows'][0]['version']);
+        // now it's sure version is a different number than original, so set it to original to allow easy comparison
+        $rollbackedConfigurationArray['rows'][0]['version'] = $originalConfigurationArray['rows'][0]['version'];
```

Comment:
Škoda že nejde srozumitelně vytvořit nové pole bez určitých klíčů aby se to nemuselo mutovat...
Mohla by to být assert metoda, ale zatím jsme  ji co vím nepotřebovali. Asi to bude tím že spíš vyjmenováváme všechny klíče které chceme porovnat.

Reply:
Jakože `$this->assertSameArrayExcept($expected, $actual, ['id']);`

Reply:
jo něco takového jsem myslel `$this->assertSameArrayExceptKeys($expected, $actual, ['id']);`

Reply:
Upravil jsem a přidal tu custom assertion